tests/integration/api/queries/orderById/orderById.test.js

```diff
@@ -0,0 +1,84 @@
+import Factory from "/imports/test-utils/helpers/factory";
+import TestApp from "/imports/test-utils/helpers/TestApp";
+// import { getAnonymousAccessToken } from "/imports/plugins/core/orders/server/no-meteor/util/anonymousToken";
+
+const mockShopId = "integ-test-shop-id";
+const opaqueShopId = "cmVhY3Rpb24vc2hvcDppbnRlZy10ZXN0LXNob3AtaWQ="; // reaction/order:integ-test-shop-id
+const shopName = "Test Shop";
+
+const orderId = "integ-test-order-id";
+const opaqueOrderId = "cmVhY3Rpb24vb3JkZXI6aW50ZWctdGVzdC1vcmRlci1pZA=="; // reaction/order:integ-test-order-id
+
+const accountId = "integ-test-account-id";
+const opaqueAccountId = "cmVhY3Rpb24vYWNjb3VudDppbnRlZy10ZXN0LWFjY291bnQtaWQ="; // reaction/account:integ-test-account-id
+
+const mockOrdersAccount = Factory.Accounts.makeOne({
+  _id: accountId
+});
+
+const order = Factory.Order.makeOne({
+  _id: orderId,
+  referenceId: "authed-db-entry",
+  shopId: mockShopId,
+  accountId: mockOrdersAccount._id
+});
+
+/* See comment below
+const tokenInfo = getAnonymousAccessToken();
+const orderIdAnon = "integ-test-order-id-anom";
+const opaqueOrderIdAnon = "cmVhY3Rpb24vb3JkZXI6aW50ZWctdGVzdC1vcmRlci1pZC1hbm9t"; // reaction/order:integ-test-order-id-anom
+
+const orderAnon = Factory.Order.makeOne({
+  _id: orderIdAnon,
+  referenceId: "anon-db-entry",
+  shopId: mockShopId,
+  anonymousAccessTokens: [{ createdAt: tokenInfo.createdAt, hashedToken: tokenInfo.hashedToken }]
+});
+*/
+
+const orderByIdQuery = `query ($id: ID!, $shopId: ID!, $token: String) {
+  orderById(id: $id, shopId: $shopId, token: $token) {
+    account {
+      _id
+    }
+    shop {
+      _id
+      name
+    }
+  }
+}`;
+
+let testApp;
+let query;
+beforeAll(async () => {
+  testApp = new TestApp();
+  await testApp.start();
+
+  query = testApp.query(orderByIdQuery);
+
+  await testApp.createUserAndAccount(mockOrdersAccount);
+  await testApp.insertPrimaryShop({ _id: mockShopId, name: shopName });
+});
+
+afterAll(() => testApp.stop());
+
+test("get account order success", async () => {
+  await testApp.collections.Orders.insertOne(order);
+  await testApp.setLoggedInUser(mockOrdersAccount);
+  const result = await query({ id: opaqueOrderId, shopId: opaqueShopId, token: null });
+  expect(result.orderById.account._id).toBe(opaqueAccountId);
+  expect(result.orderById.shop._id).toBe(opaqueShopId);
+  expect(result.orderById.shop.name).toBe(shopName);
+});
+
+/*
+TODO: I am getting a null result with this query, so this test is failing.
+Need to find out why.
```

Comment:
I see you are passing in `token: null`. Did you try `token: tokenInfo.token` and are still getting null?

Reply:
@aldeed Yes, I tried `tokenInfo.token` as well and still getting null?

Reply:
@bt3gl I ran this and I see the issue now. When you call `testApp.setLoggedInUser`, it persists for all tests in this same file, so it's looking for an account order again rather than an anonymous order.

You can fix by calling `await testApp.clearLoggedInUser();`, either at the top of your second test or in a `beforeEach`.

After you do that, you'll get a different error. To fix that error, add `accountId: null` in your `makeOne` call for `orderAnon`. If you don't explicitly set the field to `null`, the factory sets it to default string data "mockAccountId", which makes it think it's an account order.

Then it should pass!